OrbitGl/Batcher.h

```diff
@@ -64,32 +78,40 @@ class Batcher {
   Batcher() : batcher_id_(PickingID::BatcherId::TIME_GRAPH) {}
 
   void AddLine(const Line& line, const Color* colors,
-               PickingID::Type picking_type, void* user_data = nullptr);
+               PickingID::Type picking_type, 
+               std::shared_ptr<PickingUserData> user_data = nullptr);
   void AddLine(const Line& line, Color color, PickingID::Type picking_type,
-               void* user_data = nullptr);
+               std::shared_ptr<PickingUserData> user_data = nullptr);
   void AddLine(Vec2 from, Vec2 to, float z, Color color,
-               PickingID::Type picking_type, void* user_data = nullptr);
+               PickingID::Type picking_type,
+               std::shared_ptr<PickingUserData> user_data = nullptr);
```

Comment:
you might want to create 2 function - one for the case with empty userdata.

Reply:
Can you elaborate why you'd prefer that?
The AddLine e.g. has 3 overloads that all would need to be duplicated, I find a lot of overloads to be more confusing than defaults in those cases.

Reply:
because there is no nulls for r-values.

Reply:
Unchanged, because now we're using std::unique_ptr<>